pkg/apiserver/common/common.go

```diff
@@ -80,14 +74,20 @@ func Register(r *gin.RouterGroup, s *Service) {
 // @Router /common/pods [post]
 // @Failure 500 {object} utils.APIError
 func (s *Service) listPods(c *gin.Context) {
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
+		return
+	}
+
 	exp := &core.SelectorInfo{}
 	if err := c.ShouldBindJSON(exp); err != nil {
 		c.Status(http.StatusBadRequest)
 		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
 		return
 	}
 	ctx := context.TODO()
-	filteredPods, err := pkgutils.SelectPods(ctx, s.kubeCli, s.reader, exp.ParseSelector())
+	filteredPods, err := pkgutils.SelectPods(ctx, kubeCli, nil, exp.ParseSelector())
```

Comment:
use nil doesn't matter, because the kubeCli will not read from cache

Reply:
we could do `SelectPods(ctx, kubeCli, kubeCli, exp.ParseSelector())`, or when the selector is `fieldSelector` it could not successfully do `List`
https://github.com/chaos-mesh/chaos-mesh/blob/6afb841b02df35a03bbca84419e70cb95eb8c8e1/pkg/utils/selector.go#L140-L150

Reply:
Well, we could not use `SelectPods(ctx, kubeCli, kubeCli, selector)` here, because the implementation of `kubeCli` actually is a `client.DelegatingClient`. `DelegatingClient` contains a cache and require an index(not exist if we manually create one for fieldSelector) while querying with fieldSelector. This issue is considered in https://github.com/chaos-mesh/chaos-mesh/issues/878#issuecomment-688841414.

The reason why you could use this in `chaosctl` is your client is a pure `client.Client` without cache.

Reply:
My bad. Didn't noticed the change in `SelectPods`. Again thanks for the explanation!

Reply:
thank you all for review, I will update this PR after the test passed in branch `dashboard_token_test `, and then welcome to review this pr again. 😃